### PR TITLE
Set initial TxReceipt accesses to be writes instead of reads

### PR DESCRIPTION
**Pull request by z2trillion**
Also adds cumulative gas used to the block context instead of storing it as a map.

### CODE REVIEW

File: bus-mapping/src/evm/opcodes.rs [hunk 1]
```diff
@@ -438,43 +434,39 @@ pub fn gen_end_tx_ops(
     )?;
 
     // handle tx receipt tag
-    state.tx_receipt_read(
+    state.tx_receipt_write(
```
Comment (DreamWuGit) [inline bus-mapping/src/evm/opcodes.rs:437]: if change to write type, pls. update spec accordingly.

**Review by lispc — APPROVED** [review]

File: bus-mapping/src/evm/opcodes.rs (hunk 1 above)
Comment (z2trillion) [reply bus-mapping/src/evm/opcodes.rs:437]: can you take a look at https://github.com/privacy-scaling-explorations/zkevm-specs/pull/234 please? It won't let me add you as a reviewer.

**Comment by z2trillion** [comment]
Waiting to merge https://github.com/privacy-scaling-explorations/zkevm-specs/pull/234 before this.

File: bus-mapping/src/evm/opcodes.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/evm/opcodes.rs:437]: will have a look !

Comment (DreamWuGit) [reply bus-mapping/src/evm/opcodes.rs:437]: approved the spec change.

**Review by DreamWuGit — APPROVED** [review]
LGTM!

**Pull request merged**